libeos-payg/service.c

```diff
@@ -184,15 +253,35 @@ manager_new_cb (GObject      *source_object,
 {
   g_autoptr(GTask) task = G_TASK (user_data);
   EpgService *self = EPG_SERVICE (g_task_get_source_object (task));
+  g_autoptr(EpgProvider) provider = NULL;
   g_autoptr(GError) local_error = NULL;
 
-  self->provider = epg_manager_new_finish (result, &local_error);
-  if (self->provider == NULL)
+  provider = epg_manager_new_finish (result, &local_error);
+  if (provider == NULL)
     {
       g_task_return_error (task, g_steal_pointer (&local_error));
       return;
     }
 
+  epg_service_startup_complete (self, task, provider);
+}
+
+/**
+ * epg_service_startup_complete:
+ * @self:
+ * @task: (transfer none):
+ * @provider: (transfer full):
+ *
+ */
+static void
+epg_service_startup_complete (EpgService  *self,
+                              GTask       *task,
+                              EpgProvider *provider)
+{
+  g_autoptr(GError) local_error = NULL;
+
+  self->provider = provider;
```

Comment:
Nitpick: Might make things a little less likely to be broken in future to add a comment like `/* Take ownership */` above this assignment.